pilot/pkg/model/sidecar.go

```diff
@@ -199,17 +199,9 @@ func DefaultSidecarScopeForNamespace(ps *PushContext, configNamespace string) *S
 	// this config namespace) will see, identify all the destinationRules
 	// that these services need
 	for _, s := range out.services {
-		// In some scenarios, there may be multiple Services defined for the same hostname due to ServiceEntry allowing
-		// arbitrary hostnames. In these cases, we want to pick the first Service, which is the oldest. This ensures
-		// newly created Services cannot take ownership unexpectedly.
-		// However, the Service is from Kubernetes it should take precedence over ones not. This prevents someone from
-		// "domain squatting" on the hostname before a Kubernetes Service is created.
-		// This relies on the assumption that
-		if existing, f := out.servicesByHostname[s.Hostname]; f &&
-			!(existing.Attributes.ServiceRegistry != provider.Kubernetes && s.Attributes.ServiceRegistry == provider.Kubernetes) {
+		if !out.addService(s) {
 			continue
 		}
-		out.servicesByHostname[s.Hostname] = s
```

Comment:
there was no port merging if they are in same nses

Reply:
So before with sidecar and without you got merging or not?

Reply:
with or without the behavior is opposite. After git blame, these two mergings are not implemented at the same time.

Reply:
Here for the default sidecarscope, no merge at all. 
And the below for sidecar, there is.

Reply:
So in the before case, we did not _merge_, but we did actually keep the service. Note we don't filter `out.services`, but only out.servicesByHostname - which seems a bit weird.

So we would actually end up generating for the second Service, but not EDS which is super weird:
```
$ ik pc c shell-6874dcdd7f-htgw2 | rg shell
shell.default.svc.cluster.local                          9087      -          outbound      EDS
shell.default.svc.cluster.local                          27018     -          outbound      EDS
$ ik pc ep shell-6874dcdd7f-htgw2 | rg shell
10.244.0.53:9087                 HEALTHY     OK                outbound|9087||shell.default.svc.cluster.local
```